pkg/operator/etcdmemberipmigrator/etcdmemberipmigrator.go

```diff
@@ -152,6 +156,51 @@ func (c *EtcdMemberIPMigrator) reconcileMembers() error {
 	return nil
 }
 
+// getRequiredPeerListFromDNS finds the member IP using a reverse DNS lookup.
+// This approach is very heavy with dns requests but this should almost never
+// happen and this allows us to re-use the code that we have already for DNS.
+func (c *EtcdMemberIPMigrator) getRequiredPeerListFromDNS(member *etcdserverpb.Member) ([]string, error) {
+	etcdInfo, err := c.getEtcdInfo()
+	if err != nil {
+		return nil, err
+	}
+	return getRequiredPeerList(etcdInfo, member.PeerURLs)
+}
+
+// getRequiredPeerListFromFallback tries to map a member to an IP address by
+// exploiting naming conventions. For example, on Azure IPI, we know the member
+// names will be a function of the node name, and so we can find the node based
+// on the member name and use the node's IP without a DNS query.
+func (c *EtcdMemberIPMigrator) getRequiredPeerListFromFallback(member *etcdserverpb.Member) ([]string, error) {
+	infra, err := c.infrastructureLister.Get("cluster")
+	if err != nil {
+		return nil, err
+	}
+	if infra.Status.PlatformStatus == nil {
+		return nil, fmt.Errorf("no reliable platform metadata is available")
+	}
+
+	switch infra.Status.PlatformStatus.Type {
+	case configv1.AzurePlatformType:
```

Comment:
I assume this works too, but we have used the following code in the past to match Azure:
https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/etcdenvvar/etcd_env.go#L226-L231